WebAppForSolocoProject/Services/OwnerData.cs

```diff
@@ -76,49 +82,81 @@ private List<Owner> CheckForChildOwners(List<Owner> owners)
         private void RemoveStringThatIsNotPath(List<Owner> owners)
         {            
             foreach (var owner in owners)
-            {
+            {                
                 List<string> updatedPaths = new List<string>();
-                foreach (var path in owner.Paths)
+                foreach (var path in owner.Config)
                 {
                     if (path.Contains("FTP3rdparty"))
+                    {
                         updatedPaths.Add(path);
+                        if(path.Contains("SourceFolder"))
+                        {
+                            if (path.Contains("SD") && !owner.QualityFolders.Any(q=>q==Quality.SD))
```

Comment:
instead of "HD" and "SD" you can also use Quality enum. It could be foreach for enum values which check if current path
satisfied some conditions.